core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -123,20 +133,25 @@ trait WorkflowSubmission extends FutureSupport with LazyLogging with MethodWiths
     }
   }
 
-  def reserveWorkflowBatch(dataAccess: DataAccess, activeCount: Int)(implicit executionContext: ExecutionContext): ReadWriteAction[Seq[WorkflowRecord]] = {
+  def reserveWorkflowBatch(dataAccess: DataAccess, activeCount: Int)(implicit executionContext: ExecutionContext): ReadWriteAction[(Seq[WorkflowRecord], Option[SubmissionRecord], Option[WorkspaceRecord])] = {
     if (activeCount > maxActiveWorkflowsTotal) {
       logger.warn(s"There are $activeCount active workflows which is beyond the total active cap of $maxActiveWorkflowsTotal. Workflows will not be submitted.")
-      DBIO.successful(Seq.empty)
+      DBIO.successful((Seq.empty, None, None))
     } else {
       for {
         // we exclude workflows submitted by users that have exceeded the max workflows per user
         excludedUsers <- dataAccess.workflowQuery.listSubmittersWithMoreWorkflowsThan(maxActiveWorkflowsPerUser, WorkflowStatuses.runningStatuses)
         workflowRecs <- dataAccess.workflowQuery.findQueuedWorkflows(excludedUsers.map { case (submitter, count) => submitter }, SubmissionStatuses.terminalStatuses :+ SubmissionStatuses.Aborting).take(batchSize).result
         reservedRecs <- if (workflowRecs.isEmpty) {
-            DBIO.successful(workflowRecs)
+            DBIO.successful((Seq.empty, None, None))
           } else {
-            //they should also all have the same submission ID
-            DBIO.successful(workflowRecs.filter(_.submissionId == workflowRecs.head.submissionId))
+            // they should also all have the same submission ID
+            val submissionId = workflowRecs.head.submissionId
+            val filteredWorkflowRecs = workflowRecs.filter(_.submissionId == submissionId)
+            for {
+              submissionRecOpt <- dataAccess.submissionQuery.findById(submissionId).result.map(_.headOption)
```

Comment:
I like `headOption` for safety generally, but these are foreign key relations so Rawls should probably complain more loudly when this assumption fails